lib/OrdinaryDiffEqLinear/src/algorithms.jl

```diff
@@ -1,34 +1,76 @@
 # Linear Methods
 
-for Alg in [
-    :MagnusMidpoint,
-    :MagnusLeapfrog,
-    :LieEuler,
-    :MagnusGauss4,
-    :MagnusNC6,
-    :MagnusGL6,
-    :MagnusGL8,
-    :MagnusNC8,
-    :MagnusGL4,
-    :RKMK2,
-    :RKMK4,
-    :LieRK4,
-    :CG2,
-    :CG3,
-    :CG4a
-]
-    @eval struct $Alg <: OrdinaryDiffEqLinearExponentialAlgorithm
+for (Alg, Description, Ref) in [
+    (:MagnusMidpoint, "Second order Magnus Midpoint method.", "ref TBD"),
+    (:MagnusLeapfrog, "Second order Magnus Leapfrog method.", "ref TBD"),
+    (:LieEuler, "description", "ref TBD"),
+    (:MagnusGauss4, "Fourth order Magnus method approximated using a two stage Gauss quadrature.", "ref TBD"),
+    (:MagnusNC6, "Sixth order Magnus method approximated using Newton-Cotes quadrature.", "ref TBD"),
+    (:MagnusGL6, "Sixth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:MagnusGL8, "Eighth order Magnus method approximated using Newton-Cotes quadrature.", "ref TBD"),
+    (:MagnusNC8, "Eighth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:MagnusGL4, "Fourth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:RKMK2, "Second order Runge–Kutta–Munthe-Kaas method.", "ref TBD"),
+    (:RKMK4, "Fourth order Runge–Kutta–Munthe-Kaas method.", "ref TBD"),
+    (:LieRK4, "Fourth order Lie Runge-Kutta method.", "ref TBD"),
+    (:CG2, "Second order Crouch–Grossman method.", "ref TBD"),
+    (:CG3, "Third order Crouch-Grossman method.", "ref TBD"),
+    (:CG4a, " Fourth order Crouch-Grossman method.", "ref TBD")]
+    @eval begin
+        @doc generic_solver_docstring($Description,
+            $(string(Alg)),
+            "Semilinear ODE solver",
+            $Ref,
+            """
+            - `krylov`: TBD
+            - `m`: TBD
+            - `iop`: TBD
```

Comment:
Not sure if can copy the keyword description of LinearExponential.
krylov definitely takes on different values.

Reply:
I copied them from ODEExponentialRK. Reopen if that is not accurate.